target_chains/ethereum/contracts/contracts/pyth/PythGovernance.sol

```diff
@@ -190,4 +201,44 @@ abstract contract PythGovernance is
 
         emit ValidPeriodSet(oldValidPeriod, validTimePeriodSeconds());
     }
+
+    function setWormholeAddress(
+        SetWormholeAddressPayload memory payload,
+        bytes memory encodedVM
+    ) internal {
+        address oldWormholeAddress = address(wormhole());
+        setWormhole(payload.newWormholeAddress);
+
+        // We want to verify that the new wormhole address is valid, so we make sure that it can
+        // parse and verify the same governance VAA that is used to set it.
+        (IWormhole.VM memory vm, bool valid, ) = wormhole().parseAndVerifyVM(
+            encodedVM
+        );
+
+        if (!valid) revert PythErrors.InvalidGovernanceMessage();
+
+        if (!isValidGovernanceDataSource(vm.emitterChainId, vm.emitterAddress))
+            revert PythErrors.InvalidGovernanceMessage();
+
+        if (vm.sequence != lastExecutedGovernanceSequence())
+            revert PythErrors.InvalidGovernanceMessage();
+
+        GovernanceInstruction memory gi = parseGovernanceInstruction(
+            vm.payload
+        );
+
+        if (gi.action != GovernanceAction.SetWormholeAddress)
+            revert PythErrors.InvalidGovernanceMessage();
+
+        // Purposefully, we don't check whether the chainId is the same as the current chainId because
+        // we might want to change the chain id of the wormhole contract.
+
+        SetWormholeAddressPayload
```

Comment:
is this necessary? We already called `parseSetWormholeAddressPayload(gi.payload)` in `executeGovernanceInstruction`

Reply:
Same as above.